lib/features/library/widgets/section_widget.dart

```diff
@@ -72,25 +76,107 @@ class _SectionWidgetState extends State<SectionWidget>
+            ),
+          ),
+          minLeadingWidth: 0,
+          // Add pop up menu with option for removing the collection
+          trailing: PopupMenuButton(
+            offset: const Offset(0, kToolbarHeight),
+            icon: const Icon(Icons.more_vert),
+            itemBuilder: (context) {
+              return <PopupMenuEntry>[
+                PopupMenuItem(
+                  onTap: () async {
+                    final bool? shouldDelete = await Future<bool?>.delayed(
+                      const Duration(),
+                      () => showDialog<bool>(
+                        context: context,
+                        builder: (BuildContext context) {
+                          return AlertDialog(
+                            title: const Text('Remove Collection'),
+                            content: const Text(
+                                'Are you sure you want to delete this collection?'),
+                            actions: [
+                              TextButton(
+                                onPressed: () {
+                                  Navigator.of(context).pop(false);
+                                },
+                                child: const Text('CANCEL'),
+                              ),
+                              TextButton(
+                                onPressed: widget.section.header != 'Default'
+                                    ? () => Navigator.of(context).pop(true)
+                                    : null,
+                                child: Text(
+                                  'REMOVE',
+                                  style: DefaultTextStyle.of(context)
+                                      .style
+                                      .copyWith(
+                                        fontWeight: FontWeight.w500,
+                                        color: widget.section.header !=
+                                                'Default'
+                                            ? Colors.redAccent
+                                            : Theme.of(context).disabledColor,
+                                      ),
+                                ),
+                              ),
+                            ],
+                          );
+                        },
+                      ),
+                    );
+                    if (shouldDelete == null || !shouldDelete) return;
+                     final collection = AppCollection(
+                        id: widget.section
+                            .getItems()[0]
+                            .collectionIds
+                            .toList()[0],
+                        userId: widget.section.getItems()[0].userId,
+                        name: widget.section.header);
+                    final failure = await ref
+                        .read(libraryViewControllerProvider.notifier)
+                        .removeBookCollection(collection);
```

Comment:
It might be easier to create a new method in LibraryViewController that takes the collection name or id (would need to add id to CollectionSection) and uses that to delete it